cli/azd/pkg/infra/provisioning/manager.go

```diff
@@ -19,41 +20,79 @@ type Manager struct {
 	console     input.Console
 }
 
+// Infrastructure deployment options
+type DeployOptions struct {
+	// Whether or not the session supports interactive user input
+	Interactive bool
+}
+
+// Infrastructure destroy options
+type DestroyOptions struct {
+	// Whether or not the session supports interactive user input
+	Interactive bool
+	// Whether or not to force the deletion of resources without prompting the user
+	Force bool
+	// Whether or not to purge any key vaults associated with the deployment
+	Purge bool
+}
+
 // Prepares for an infrastructure provision operation
 func (m *Manager) Preview(ctx context.Context, interactive bool) (*PreviewResult, error) {
 	previewResult, err := m.preview(ctx, interactive)
 	if err != nil {
 		return nil, err
 	}
 
-	updated, err := m.ensureParameters(ctx, &previewResult.Preview)
-	if err != nil {
-		return nil, err
-	}
+	return previewResult, nil
+}
 
-	if updated {
-		if err := m.provider.UpdatePlan(ctx, previewResult.Preview); err != nil {
-			return nil, fmt.Errorf("updating deployment parameters: %w", err)
-		}
+// Gets the latest deployment details for the specified scope
+func (m *Manager) GetDeployment(ctx context.Context, scope Scope, interactive bool) (*DeployResult, error) {
+	var deployResult *DeployResult
+
+	spinner := spin.NewSpinner("Retrieving Azure Deployment")
+	err := spinner.Run(func() error {
+		queryTask := m.provider.GetDeployment(ctx, scope)
+
+		go func() {
+			for progressReport := range queryTask.Progress() {
+				if interactive {
+					spinner.Title(progressReport.Message)
+				}
+			}
+		}()
 
-		if err := m.env.Save(); err != nil {
-			return nil, fmt.Errorf("saving env file: %w", err)
+		go monitorInteraction(spinner, queryTask.Interactive())
+
+		result, err := queryTask.Await()
+		if err != nil {
+			return err
 		}
+
+		deployResult = result
+
+		return nil
+	})
+
+	if err != nil {
+		return nil, fmt.Errorf("error deploying infrastructure: %w", err)
 	}
 
-	return previewResult, nil
+	m.console.Message(ctx, "Azure env refresh completed")
```

Comment:
Is this going to output something to `stdout` if `!interactive`? (it should not)

Reply:
Updated so the `console` looks at the current formatter and determines whether to log to the current `io.Writer` or just to the log.  

So for JSON format, any `console.Message(...)` call will only ever write to the `log` and not `stdout`